### PR TITLE
Publish custom rancher chart with operator

### PR DESCRIPTION
**Pull request by alexander-demicev**
Publish custom rancher chart with operator. Currently, there is no way to test changes in operator CRDs before tagging a release and merging an update PR in rancher/rancher. This PR adds a step to the nightly workflow to update rancher dependencies and build a chart based on that.

Fixes: https://github.com/rancher/highlander/issues/20
Fixes: https://github.com/rancher/highlander/issues/18
Fixes: https://github.com/rancher/highlander/issues/21

### CODE REVIEW

**Comment by salasberryfin** [comment]
Just nitpicking a bit, sorry. LGTM, though.

I understand this is intended to publish updates to current Rancher version (in this case 2.8) and back-ports to other currently maintained versions (as we've been doing lately) will be done "manually". Is this correct?

**Comment by salasberryfin** [comment]
Looks like this needs a manual rebase.

**Comment by salasberryfin** [comment]
Sorry, I know I'm being very picky but, can we update the typos in the name of the operator (`aks` to `eks`)?

File: .github/workflows/update-rancher-charts.yaml [hunk 1]
```diff
@@ -2,19 +2,15 @@ name: Update EKS Operator in rancher/charts
...
-        description: "Branch to use for GitHub action workflow"
-        required: true
-        default: "master"
       charts_ref:
         description: "Submit PR against the following rancher/charts branch (e.g. dev-v2.7)"
         required: true
-        default: "dev-v2.7"
-      prev_eks_operator:
+        default: "dev-v2.8"
+      prev_aks_operator:
```
Comment (richardcase) [inline .github/workflows/update-rancher-charts.yaml:9]:
```suggestion
      prev_eks_operator:
```

File: .github/workflows/update-rancher-charts.yaml [hunk 2]
```diff
@@ -2,19 +2,15 @@ name: Update EKS Operator in rancher/charts
...
         description: "Previous EKS operator version (e.g. 1.1.0-rc2)"
         required: true
         default: ""
-      new_eks_operator:
+      new_aks_operator:
```
Comment (richardcase) [inline .github/workflows/update-rancher-charts.yaml:13]:
```suggestion
      new_eks_operator:
```

File: .github/workflows/update-rancher-charts.yaml [hunk 3]
```diff
@@ -32,35 +28,15 @@ on:
...
-            cc @rancher/highlander
-          branch-suffix: timestamp
-          base: ${{github.event.inputs.charts_ref}}
-          path: ./charts/
+  update-rancher-chart:
+    uses: rancher-sandbox/highlander-reusable-workflows/.github/workflows/update-rancher-charts.yaml@main
+    with:
+      charts_ref: ${{ github.event.inputs.charts_ref }}
+      operator: eks-operator
+      prev_operator_version: ${{ github.event.inputs.prev_aks_operator }}
```
Comment (richardcase) [inline .github/workflows/update-rancher-charts.yaml:36]:
```suggestion
      prev_operator_version: ${{ github.event.inputs.prev_eks_operator }}
```

File: .github/workflows/update-rancher-charts.yaml [hunk 4]
```diff
@@ -32,35 +28,15 @@ on:
...
+      new_operator_version: ${{ github.event.inputs.new_aks_operator }}
```
Comment (richardcase) [inline .github/workflows/update-rancher-charts.yaml:37]:
```suggestion
      new_operator_version: ${{ github.event.inputs.new_eks_operator }}
```

File: .github/workflows/update-rancher-dep.yaml [hunk 5]
```diff
@@ -2,57 +2,19 @@ name: Update EKS operator in rancher/rancher
...
-        required: true
-        default: "master"
       rancher_ref:
         description: "Submit PR against the following rancher/rancher branch (e.g. release/v2.7)"
-        required: true
-        default: "release/v2.7"
+        default: "release/v2.8"
       new_eks:
-        description: "New EKS operator version (e.g. 1.1.0-rc2), don't include the 'v'"
+        description: "New AKS operator version (e.g. 1.1.0-rc2), don't include the 'v'"
```
Comment (richardcase) [inline .github/workflows/update-rancher-dep.yaml:9]:
```suggestion
        description: "New EKS operator version (e.g. 1.1.0-rc2), don't include the 'v'"
```

**Review by salasberryfin — APPROVED** [review]

**Pull request merged**